### ISSUE TITLE
Known Camera Pose

### ISSUE THREAD

**Issue by feiyu12138** [facebookresearch/vggt#5]
Congrats on super accurate and efficient model!

Could I please ask if it's possible to support known camera parameter situtation?

**Comment by jytime** [comment]
Hi Feiyu,

Thanks for your interest! Supporting known camera parameters isn’t currently within the main scope of this repository. However, based on our preliminary experiments, it can be easily achieved with minor fine-tuning using LoRA or similar techniques.

Best,
Jianyuan

**Comment by erwin-wu-x** [comment]
Hi Jianyuan,

Thank you for your great contributions to the community!

Could you please provide more details on how to use LoRA or other methods to fine-tune a model with known camera poses? I apologize if this seems like a "simple" question, but I am new to LoRA and would greatly appreciate any guidance.

Additionally, do you have any plans to share the weights or provide simple code to perform something like SfM with known poses?

Thank you again for your work!

**Comment by jytime** [comment]
Hi @erwin-wu-x 

Basically you just need to a way to adjust the attention layers by your input information. The basic idea is to freeze most layers of the model and only add a residual to those layers.

For example, here is an explanation for lora (chinese) https://zhuanlan.zhihu.com/p/639229126

**Comment by Livioni** [comment]
Hi @jytime 

One step further to this question, if I use the known camera parameter (truth intrinsics and extrinsics) to fine-tune the model by LoRA, should I just inject the truth camera parameters by adding the camera token and remove the Camera Head? It there any advice on it?

Thank you for your great work!

**Comment by jytime** [comment]
Hi @Livioni, I feel you should not inject the GT camera into the camera token only. Instead, the GT camera information should be added to the whole tokens/attention layers. But probably injecting into camera tokens could be a good start.

**Comment by missTL** [comment]
May I ask if it is now supported to use the known intrinsic parameters and poses of the camera?

**Comment by PeaceNeil** [comment]
Hi @Livioni @erwin-wu-x @jytime,

Thanks for the great work and discussion. I tried to modify the fine-tuning pipeline with Lora. The parameters for tuning the camera and depth heads were reduced from 248.8M to 1.9M, but the required GPU memory increased instead. 

Initially, I thought it was because the optimizer's parameter included the frozen parts, but after editing, the memory usage remained unchanged. I suspect that the gradients and activation values being stored are still occupying a large amount of GPU memory.

Here are some questions:  
- If only the camera and depth heads are being trained, do the aggregator's parameters need to be included in the optimizer? 
- Why does including or excluding frozen parameters in the optimizer not have an impact on memory usage?  
- Is there a way to use LoRA and reduce memory usage?

Thanks for any help!

**Comment by LanesraL** [comment]
Hi @Livioni 
I am going to fine-tune the model with camera parameter. I would like to ask what kind of performance improvements you have achieved by doing this. 
By the way, It's great to see that you are also from BIT. May I have the honor to add your WeChat to discuss further?

Thanks for any help!

**Comment by Livioni** [comment]
Hi ! @LanesraL, nice to hear that. Come and add me on WeChat (Livion2018) ASAP. lol

**Comment by isxiaohe** [comment]
Hi @LanesraL 
Have you tried fine tuning this model? Is there any progress?

**Comment by LanesraL** [comment]
Hi @isxiaohe 
I only attempted to fine-tune VGGT with pose input added, and only conducted experiments on a small dataset, with unsatisfactory results. 
Now I have stopped this line of work due to the release of MapAnything.
But we can still communicate.
